misc/python/materialize/parallel_workload/parallel_workload.py

```diff
@@ -0,0 +1,302 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import argparse
+import datetime
+import os
+import random
+import sys
+import threading
+import time
+from collections import Counter, defaultdict
+from typing import DefaultDict, List, Optional, Type
+
+import pg8000
+
+from materialize.mzcompose import Composition
+from materialize.parallel_workload.action import (
+    Action,
+    CancelAction,
+    KillAction,
+    ddl_action_list,
+    dml_nontrans_action_list,
+    fetch_action_list,
+    read_action_list,
+    write_action_list,
+)
+from materialize.parallel_workload.database import Database
+from materialize.parallel_workload.executor import Executor, initialize_logging
+from materialize.parallel_workload.worker import Worker
+
+SEED_RANGE = 1_000_000
+REPORT_TIME = 10
+
+
+def run(
+    host: str,
+    port: int,
+    system_port: int,
+    seed: str,
+    runtime: int,
+    complexity: str,
+    scenario: str,
+    num_threads: Optional[int],
+    composition: Optional[Composition],
+) -> None:
+    num_threads = num_threads or os.cpu_count() or 10
+    random.seed(seed)
+
+    print(
+        f"--- Running with: --seed={seed} --threads={num_threads} --runtime={runtime} --complexity={complexity} --scenario={scenario} (--host={host} --port={port})"
+    )
+    initialize_logging()
+
+    system_conn = pg8000.connect(
+        host=host, port=system_port, user="mz_system", database="materialize"
+    )
+    system_conn.autocommit = True
+    with system_conn.cursor() as cur:
+        # TODO: Shuffle settings which should always work
```

Comment:
How does that list differ from the list in mzcompose/services.py

Reply:
It has two functions:
1. I often ran bin/parallel-workload locally for faster turnaround times than starting up mzcompose; noticed that I didn't run into some issues I saw in mzcompose, like RBAC panic
2. I wanted to extend this with all the available options (which don't change results) and have a setting to shuffle them

Reply:
Done.